src/main/java/org/edumips64/utils/cli/DineroCommand.java

```diff
@@ -0,0 +1,54 @@
+package org.edumips64.utils.cli;
+
+import org.edumips64.core.CPU;
+import org.edumips64.utils.CurrentLocale;
+import org.edumips64.utils.io.LocalWriterAdapter;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.ParentCommand;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Option;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@Command(name = "dinero", resourceBundle = "CliMessages")
+public class DineroCommand implements Runnable {
+
+    private static final String DINERO_SUFFIX = ".xdin";
+
+    @ParentCommand
+    private Cli cli;
+
+    @Option(names = {"-h", "--help"}, usageHelp = true, descriptionKey = ".help.usage.description")
+    private boolean help;
+
+    @Parameters(arity = "1", descriptionKey = "dinero.param", paramLabel = "target_file")
+    private String file;
+
+    @Override
+    public void run() {
+        try (PrintWriter pw = new PrintWriter(appendDineroSuffix(file))){
+            if (cli.getCPU().getStatus() == CPU.CPUStatus.HALTED) {
+                cli.getDinero().writeTraceData(new LocalWriterAdapter(pw));
+                pw.flush();
+            } else {
+                System.out.println(CurrentLocale.getString("CLI.DINERO.CANT.WRITE"));
+            }
+        } catch (FileNotFoundException fnfe) {
+            System.out.println(CurrentLocale.getString("CLI.FILE.NOT.FOUND"));
+        } catch (IOException ioe) {
+            System.out.println(CurrentLocale.getString("CLI.IO.ERROR"));
+        } catch (Exception e) {
+            Cli.printErrorMessage(e);
+        }
+    }
+
+    private String appendDineroSuffix(String s) {
```

Comment:
Why is this needed? I think it might be surprising for the user if they ask the simulator to write the tracefile to `foo.txt` and we write to `foo.txt.xdin`.

Reply:
I did this because in the GUI when the user wants to write out a tracefile it is presented with that suffix in the file dialog. Do you want me to change or leave as is? I realize in the GUI the user can change the extension in the file dialog but here I would imagine a CLI user is more saavy and could just rename the file if they wanted a different name.

Reply:
I understand what you mean. I do not fully agree (I'd prefer users to fully control the output file name), but I accept your choice!

Can you please echo back to the user the actual filename the code used, so that they will know whether we added the extension or not?

Reply:
Added the echo for the file name. Removed the extension, it was easier to echo without adding.